### ISSUE TITLE
HPE DL360 HPE Ethernet 1Gb 4-port 331i Adapter - NIC:  iPXE hangs on one server but not on another

### ISSUE THREAD

**Issue by Rampftl** [ipxe/ipxe#635]
I have two HPE DL360 Servers with HPE Ethernet 1Gb 4-port 331i Adapters. When I try to PXE boot with iPXE one server is working fine. The other gets stuck on iPXE initializing devices.... 
Both servers have the same BIOS Version and all Hardware Firmwares are updated to the same Version. I use ipxe.efi compiled from the latest sources.
Any idea?

**Comment by NiKiZe** [comment]
Could you try snponly.efi on both, and also intelx.efi on both (assuming this is an Intel device compatible with intelx, also post pciid)

**Comment by 9001** [comment]
Also check the iLO configuration -- it can be configured to make itself available over a regular nic-port instead of (or in addition to) the dedicated iLO port, and maybe that could cause issues when iPXE tries to swap out the drivers, so check that "Shared network" isn't enabled

**Comment by Rampftl** [comment]
> Also check the iLO configuration -- it can be configured to make itself available over a regular nic-port instead of (or in addition to) the dedicated iLO port, and maybe that could cause issues when iPXE tries to swap out the drivers, so check that "Shared network" isn't enabled

Thanks for the support. Shared network isn't enabled. But using snponly.efi did the trick :)

**Comment by Rampftl** [comment]
> Could you try snponly.efi on both, and also intelx.efi on both (assuming this is an Intel device compatible with intelx, also post pciid)

I compiled and tried snponly.efi and it worked on both machines. Thank you very much. 

Next test: I will use intelx.efi on both.

**Comment by stappersg** [comment]
> Next test: I will use intelx.efi on both.

And how did that go?


Also about five days ago:
> also post pciid

Those _PCI  identities_ can be found with the Linux tool `lspci`.   I executed that command on my laptop and then zoom in, `-v` on the _slot_, `-s`, with the network interface card:
```text
$ lspci -v -s 01:00.0
01:00.0 Ethernet controller: Broadcom Inc. and subsidiaries NetXtreme BCM57786 Gigabit Ethernet PCIe (rev 01)
	Subsystem: Acer Incorporated [ALI] NetXtreme BCM57786 Gigabit Ethernet PCIe
	Flags: bus master, fast devsel, latency 0, IRQ 18, IOMMU group 10
	Memory at b0410000 (64-bit, prefetchable) [size=64K]
	Memory at b0420000 (64-bit, prefetchable) [size=64K]
	Expansion ROM at 9fb00000 [disabled] [size=2K]
	Capabilities: <access denied>
	Kernel driver in use: tg3
	Kernel modules: tg3
```

I think that #640 also should get _pciids_.
( Do know that [how to ask smart questions](http://www.catb.org/~esr/faqs/smart-questions.html) does not approve _yes-no-questions_. )

**Comment by Rampftl** [comment]
Regarding pciid:

I tried to find the pciids with devmgmt under windows:
<img width="200" alt="srv-1-pci" src="https://user-images.githubusercontent.com/7591395/166220187-a7b00fd9-ab1b-47ed-8906-175a60d638ca.png">
<img width="202" alt="srv-2-pci" src="https://user-images.githubusercontent.com/7591395/166220183-e79c8c63-780c-4ef1-8e79-9464182a8dcc.png">
Hope this is the correct id

**Comment by Rampftl** [comment]
Intelx.efi starts on both servers but returns message "no more network devices".

**Comment by NiKiZe** [comment]
My bad in assuming it was a Intel device. This is the same as the other thread? Please keep it in text 14e4:1657 (?)

**Comment by stappersg** [comment]
> Hope this is the correct id

It is the _pci id_ that @Rampftl provided.

I had a look at both screenshots. ( I don't understand why there are two.)

> Please keep it in text 14e4:1657 (?)

I also did read `14e4:1657`, more detailed version:
```text
0200: 14e4:1657 (rev 01)
Subsystem: 22be:103c
```

https://devicehunt.com/view/type/pci/vendor/14E4/device/1657 says that `14e4` is _"Broadcom Inc. and subsidiaries"_.

**Comment by stappersg** [comment]
@Rampftl do you meanwhile know why there is a working server?

**Comment by Rampftl** [comment]
> @Rampftl do you meanwhile know why there is a working server?

not really. There is an assumption but not verified yet. One of the servers (the one which is not working with ipxe.efi) has a PCI card for printer head control installed, the other one has not. I did not get the model and vendor of the card yet. Perhaps it interferes with ipxe.efi by letting it think it is a network card.

**Comment by stappersg** [comment]
This issue was never an _iPXE_ issue. In the initial message is _"boot with iPXE one server is working fine"_.

The _"Any idea?"_ was responsed with _"provide PCI ids"_ and _"compare the two systems"_ the very same day.

Now, three weeks later, there is still no _PCI id_ of suspected card.

@Rampftl choose,  either take this issue further or close it.  Don't no keep it dangling. Do know that dangling issues drain human energy.

**Comment by NiKiZe** [comment]
> This issue was never an _iPXE_ issue. In the initial message is _"boot with iPXE one server is working fine"_.
> 
> The _"Any idea?"_ was responsed with _"provide PCI ids"_ and _"compare the two systems"_ the very same day.
> 
> Now, three weeks later, there is still no _PCI id_ of suspected card.
> 
> @Rampftl choose, either take this issue further or close it. Don't no keep it dangling. Do know that dangling issues drain human energy.

When speaking about draining energy, take a look at your own irrelevant, rude comments.

**Comment by Rampftl** [comment]
> This issue was never an _iPXE_ issue. In the initial message is _"boot with iPXE one server is working fine"_.
> 
> The _"Any idea?"_ was responsed with _"provide PCI ids"_ and _"compare the two systems"_ the very same day.
> 
> Now, three weeks later, there is still no _PCI id_ of suspected card.
> 
> @Rampftl choose, either take this issue further or close it. Don't no keep it dangling. Do know that dangling issues drain human energy.

I hoped to provide the PCIID of the network card with my two screenshots several weeks ago. Unfortunately the servers are running under WIndows and are not in my direct access. I requested a check with the owers regarding the "printer head controll card", which is a special one but it takes a while. Sorry for that.
Point is, that the servers usually are equipped with that card, as working with it is their main purpose. snponly.efi allows to pxe boot them (with or without the ominous card). So from the owners side the problems seems to be solved.
I will provide you with the list of PCI devices of both servers as soon as I get them.

**Comment by stappersg** [comment]
Let's assume that some day two lists of PCI device IDs are provided.
With `sort` and `diff` we can find easily differences.

With some luck is the only different indeed the **non** working system having a _"printer head controll card"_.

Next step would be adding the PCI-ID to the _blocklist_ of  iPXE.

```text
$ git grep -i lklist
src/include/ipxe/pccrr.h:struct peerdist_msg_getblklist {
src/include/ipxe/pccrr.h:#define peerdist_msg_getblklist_t( digestsize, count )                 \
src/include/ipxe/pccrr.h:               struct peerdist_msg_getblklist getblklist;              \
src/include/ipxe/pccrr.h:#define PEERDIST_MSG_GETBLKLIST_VERSION PEERDIST_MSG_VERSION_1_0
src/include/ipxe/pccrr.h:#define PEERDIST_MSG_GETBLKLIST_TYPE 0x00000002UL
src/include/ipxe/pccrr.h:struct peerdist_msg_blklist {
src/include/ipxe/pccrr.h:#define peerdist_msg_blklist_t( digestsize, count )                    \
src/include/ipxe/pccrr.h:               struct peerdist_msg_blklist blklist;                    \
src/include/ipxe/pccrr.h:#define PEERDIST_MSG_BLKLIST_VERSION PEERDIST_MSG_VERSION_1_0
src/include/ipxe/pccrr.h:#define PEERDIST_MSG_BLKLIST_TYPE 0x00000004UL
```

Frankly, I can't tell if those _*blklist_s are for _block using this PCI  ID_.

I think it is better to close this issue and open a wishlist item issue when there is envidence for such case.

**Comment by stappersg** [comment]
This update was written after reading _"I expect both machine to have same behaviors"_, which reminded me of this issue.
And that did remind of [_How to ask smart questions_](http://www.catb.org/~esr/faqs/smart-questions.html), that it is okay to ignore questions that don't deserve an answer.

Reason for this update is to find if mentioning #727 then that discussion gets a link to this issue. Like `https://github.com/ipxe/ipxe/issues/529#issuecomment-1114733732`, https://github.com/ipxe/ipxe/issues/529#issuecomment-1114733732, did.

And I still would like to known what hardware configuration makes iPXE getting stuck on initializing devices. So we can explore how to avoid that. :wink:

**Comment by 9001** [comment]
> And that did remind of [_How to ask smart questions_](http://www.catb.org/~esr/faqs/smart-questions.html)

Please stop
